### ISSUE TITLE
CloudCompare built successfully but shows error on running the generated application.

### ISSUE THREAD

**Issue by radhikaravi251** [CloudCompare/CloudCompare#614]
Hi,
I have successfully built the Visual Studio (2013, x64) solution for CloudCompare but when I run the generated application, it shows the error: "The application was unable to start correctly (0xc000007b). Click OK to close the application.

Could anyone please help me with this error. I checked that all the required dependencies and .dll's are the ones generated for 64 bits as well, so there is no mismatch.

Thanks,
Radhika

**Comment by dgirardeau** [comment]
Is the "standard" version working on your side? What is your graphic card?

**Comment by radhikaravi251** [comment]
Yes, the standard version downloaded as an executable functions properly on the system. It's only when building the whole thing from the source code that it shows this error.
I have a 4 gb graphics card.

**Comment by dgirardeau** [comment]
Oh you could have a old ATI card with 16Gb of memory and it would still fail ;)

Anyway, regarding the compiled version:
- what version of Qt are you using?
- have you enabled GL_QUAD_BUFFER support maybe?
- have you run it in debug mode to see where it crashes?

**Comment by radhikaravi251** [comment]
- Qt Version: 5.6.3
- Yes, I've enabled GL_QUAD_BUFFER support
- And in debug mode, it crashes even before executing any line of code. I think it's because of some .dll being incompatible or not being found, probably some system dll's. :-/

**Comment by dgirardeau** [comment]
A missing DLL won't make the program crash (but just pop an error message).

In debug mode, you should at least see "where" the program crash (see the call stack). Generally it's in the GPU driver ;).

And have you tried without GL_QUAD_BUFFER support? (because this mode requires a recent graphic card...).

Sorry to always speak about the graphic card, but 99% of crashes are due to that.

**Comment by radhikaravi251** [comment]
Yes, and that's what I'm getting: a successful build but a runtime error. Below is a screenshot:

![image](https://user-images.githubusercontent.com/31517517/32096237-d8950434-bad3-11e7-9ad1-6d5731ff2e4f.png)

Also, I'll try without the GL_QUAD_BUFFER support too.

Thanks :-)

**Comment by dgirardeau** [comment]
This is not debug mode :)

(see the 'Release' entry of the combo below 'Help'). And you have to start your program with F5.

**Comment by radhikaravi251** [comment]
Yes, it is giving me the same error both in Release and Debug mode. And yes, I did compile the 'INSTALL' project. I have attached a screenshot for the Debug mode also.
![image](https://user-images.githubusercontent.com/31517517/32121342-3c861d88-bb2a-11e7-9b8d-bea9dcdb8cdd.png)

**Comment by radhikaravi251** [comment]
UPDATE: I used dependency walker to check the dependencies. It seems like there is are some Visual Studio dll's missing, for which I need to install the Visual C++ Redistributable Packages. I'll do that and update you on whether I still encounter any issues.

**Comment by lmarsXubo** [comment]
I met same error with you and the problem still not solved. 
According to the dependency walker, the missed dlls seem to be all named as:  
API-MS-WIN-CORE-******.DLL
which seem to be the windows run time dlls.
I have already installed the Visual C++ Redistributable Packages,  the windows update as well as the  drivers for the graphics card. However, the problem still occurs.
Please tell me if you have already solve the problem.
I install the newest Cloudcompare under win7-64+ PCL 1.8.0 via vs 2013.
![default](https://user-images.githubusercontent.com/33014609/32141068-899809dc-bcb1-11e7-8498-d456c08c807e.png)

**Comment by dgirardeau** [comment]
On my side I have the same 'errors' in Dependency Walker. However it works well. It's more a limitation of Dependency Walker, not a problem with CloudCompare (you can try with other programs). For instance you are not responsible for anything that is below Kernel32.dll (which is already in the Windows kernel).

You should focus on the first (or second) level DLLs and especially the custom ones. For instance, did you compile PCL yourself or did you get a pre-compiled package? Are all the PCL DLLs next to CloudCompare.exe? Have you tried to compile and run CloudCompare without any dependency (apart Qt of course)?

**Comment by radhikaravi251** [comment]
I'm using CloudCompare with libLAS-1.8.1. I compiled the library myself for 64-bit version. I have the DLLs in the same directory as CloudCompare.exe. 
CloudCompare is running without any problems if I do not have libLAS as a dependency. 
I'm not sure why libLAS is causing problems with 64 bit. I have built it similarly for 32 bit and it works fine for other applications.
Any suggestions are welcome.
Thanks.

**Comment by dgirardeau** [comment]
What if you apply the dependency walker on the liblas DLL that is in the CloudCompare folder?

**Comment by radhikaravi251** [comment]
Good news! I got CloudCompare to run without any issues along with libLAS. I did the entire build on another system, which proceeded quite smoothly. On examining, I found that there was one liblas.dll from OSGeo4W and another from my own build of libLAS library. Using the first *.dll causes errors. However, if I replace it in the directory of CloudCompare.exe with the liblas.dll that I generate, everything works fine.
P.S. Also check for multiple versions of the same library being installed in different include directories. In this case, the order of search in directories would influence whether the correct link libraries are used or not.

Thanks a lot, Daniel @dgirardeau  for your prompt responses.
